app/models/pame_evaluation.rb

```diff
@@ -79,14 +81,17 @@ def self.generate_query(page, filter_params)
 
   def self.parse_filters(filters)
     site_ids = []
-    where_params = {sites: "", methodology: "", year: ""}
+    country_ids = []
+    where_params = {sites: "", methodology: "", year: "", iso3: ""}
     filters.each do |filter|
       options = filter["options"]
       case filter['name']
       when 'iso3'
         countries = options
         site_ids << countries.map{ |iso3| Country.find_by(iso_3: iso3).protected_areas.pluck(:id) }
         where_params[:sites] = options.empty? ? nil : "protected_area_id IN (#{site_ids.join(',')})"
```

Comment:
I think this line and line 94 should be changed slightly.
Even though the `options` variable is populated, if `sites_ids` is empty this will result into an empty SQL statement for `protected_area_id IN()`, which breaks things.
We should check against `sites_ids` itself rather then `options` in this case

Reply:
Many thanks @Levia for your feedback on this!!

Reply:
Uhm, Ferdi, for some reason I cannot reproduce this error myself. I thought it would go through all of the filters and check against the `filter['name']` being for example `iso3` and in that case there must be some `iso3` codes present in that case. But I am starting to think that if data is sent from the user which is not correct this could trigger an error. I am not sure how you got the error as I cannot reproduce that. I will try and improve this anyway.

Reply:
@drewpotter The issue shows up depending on the records in your DB.
You don't need to replicate this to spot the issue really; if, for any reason, `site_ids`is empty but `options` is not, the ```where_params[:sites] = options.empty? ? nil : "protected_area_id IN (#{site_ids.join(',')})"``` will result in:

```protected_area_id IN ()```

which is incorrect.
So I would change it like so:
```where_params[:sites] = site_ids.empty? ? nil : "protected_area_id IN (#{site_ids.join(',')})"```
as it's clear that this `where` statements depends on the ids inside `site_ids`

Reply:
Many thanks!